Experimental/src/org/sleuthkit/autopsy/experimental/textclassifier/TextClassifierTrainer.java

```diff
@@ -0,0 +1,243 @@
+        progressPanel.updateStatusLabel(NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.inProgress.text"));
+
+        ObjectStream<DocumentSample> sampleStream;
+        try {
+            sampleStream = processTrainingData(progressPanel);
+        } catch (Exception ex) {
+            new File(baseReportDir).delete();
+            return;
+        }
+
+        DoccatModel model = null;
+
+        progressPanel.setIndeterminate(true);
+        progressPanel.updateStatusLabel(NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.training.text"));
+        try {
+            model = train(TextClassifierUtils.MODEL_PATH, sampleStream);
+        } catch (IOException ex) {
+            progressPanel.complete(ReportStatus.ERROR);
+            progressPanel.updateStatusLabel(NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.noModel.text"));
+            new File(baseReportDir).delete();
+            return;
+        }
+
+        if (model == null) {
+            progressPanel.complete(ReportStatus.ERROR);
+            progressPanel.updateStatusLabel(NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.noModel.text"));
+            new File(baseReportDir).delete();
+            return;
+        }
+        try {
+            progressPanel.setIndeterminate(true);
+            progressPanel.updateStatusLabel(NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.writingModel.text") + TextClassifierUtils.MODEL_PATH);
+            TextClassifierUtils.writeModel((NaiveBayesModel) model.getMaxentModel(), TextClassifierUtils.MODEL_PATH);
+            progressPanel.complete(ReportStatus.COMPLETE);
+            progressPanel.updateStatusLabel(NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.completeModelLocation.text") + TextClassifierUtils.MODEL_PATH);
+        } catch (IOException ex) {
+            progressPanel.complete(ReportStatus.ERROR);
+            progressPanel.updateStatusLabel(NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.cannotSave.text") + TextClassifierUtils.MODEL_PATH);
+            new File(baseReportDir).delete();
+            return;
+        }
+    }
+
+    @Messages({"TextClassifierTrainer.getName.text=Text classifier model"})
+    @Override
+    public String getName() {
+        String name = NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.getName.text");
+        return name;
+    }
+
+    @Messages({"TextClassifierTrainer.getDesc.text=Machine learning model to classify which documents are notable"})
+    @Override
+    public String getDescription() {
+        String desc = NbBundle.getMessage(this.getClass(), "TextClassifierTrainer.getDesc.text");
+        return desc;
+    }
+
+    @Override
+    public String getRelativeFilePath() {
+        return "classifierTrainerReport.txt";
+    }
+
+    @Override
+    public JPanel getConfigurationPanel() {
+        configPanel = new TextClassifierTrainerConfigPanel();
```

Comment:
I believe the default GeneralReportModuleAdapter implementation of returning null is what we would want here instead of an over ride returning a panel with nothing in it.